### PR TITLE
options

### PR DESCRIPTION
**Pull request by maxiride**
Addresses #7 

Honestly I don't know how to handle an optional struct altogether, so there is a margin of improvement there either make it a variadic parameter or a pointer or something else if you want to keep consistency of it being an optional parameter.

### CODE REVIEW

File: fillpdf.go [hunk 1]
```diff
@@ -32,9 +32,17 @@ import (
 // This is a key value map.
 type Form map[string]interface{}
 
+// Options represents the options to alter the PDF filling process
+// Overwrite will overwrite any pre existing filled PDF
+// Flatten will flatten the document making the form field no longer editable
```
Comment (r0l1) [inline fillpdf.go]: Move the comments to their definitions. Each options field will have it's own comment...

File: fillpdf.go [hunk 2]
```diff
@@ -32,9 +32,17 @@ import (
...
+type Options struct {
+	Overwrite bool
+	Flatten   bool
+}
+
 // Fill a PDF form with the specified form values and create a final filled PDF file.
-// One variadic boolean specifies, whenever to overwrite the destination file if it exists.
-func Fill(form Form, formPDFFile, destPDFFile string, overwrite ...bool) (err error) {
+// The options parameter alters few aspects of the generation.
+func Fill(form Form, formPDFFile, destPDFFile string, options Options) (err error) {
```
Comment (r0l1) [inline fillpdf.go]: Make options variadic, to keep in sync with the previous API.

```go
var o Options
if len(options) > 0 {
  o = options[0]
}
```

**Review by r0l1 — COMMENTED** [review]
Looks good. Let's stick to the variadic style. Pointer is not required for such a small struct.

File: fillpdf.go (hunk 1 above)
Comment (maxiride) [reply fillpdf.go]: Done.

File: fillpdf.go (hunk 2 above)
Comment (maxiride) [reply fillpdf.go]: Implemented default values and made the parameter variadic again.

**Review by r0l1 — APPROVED** [review]

**Pull request merged**